File: 409-LongestPalindrome/409-LongestPalindrome.cpp
// Last updated: 13/07/2026, 22:20:03
class Solution {
public:
    int longestPalindrome(string s) {
        long long state = 0;
        int result = 0;
        for (const char& ch : s){
            if (state & (1LL << (ch - 'A'))){
                result += 2;
                state = state ^ (1LL << (ch - 'A'));
            }
            else {
                state |= 1LL << (ch - 'A');
            }
        }
        if (state) ++result;
        return result;
    }
};

File: 409-LongestPalindrome/409-LongestPalindrome.py
# Last updated: 13/07/2026, 22:20:09
class Solution:
    def longestPalindrome(self, s: str) -> int:
        charSet = set()
        result = 0
        for char in s:
            if (char in charSet):
                charSet.remove(char)
                result += 2
            else:
                charSet.add(char)
        
        if (charSet):
            result += 1
            
        return result